recipes/openssl/ALL/conanfile.py

```diff
@@ -533,6 +533,12 @@ def package(self):
                 with tools.chdir(os.path.join(self.package_folder, 'lib')):
                     os.rename('libssl.lib', 'libssld.lib')
                     os.rename('libcrypto.lib', 'libcryptod.lib')
+        # Old OpenSSL version family has issues with permissions.
+        # See https://github.com/conan-io/conan/issues/5831
+        if self._full_version < "1.1.0" and self.options.shared and self.settings.os != "Windows":
+            with tools.chdir(os.path.join(self.package_folder, "lib")):
+                os.chmod("libssl.so.1.0.0", 0o755)
```

Comment:
this may fail on Mac (OSX) where shared libraries have `.dylib` ext

Reply:
Yep, thanks!

Reply:
@SSE4 I've pushed modification.

Reply:
@Talkless seems like not, I can't see it

Reply:
@SSE4 strange, I see this updated commit in my clone:
https://github.com/Talkless/conan-center-index/commit/ad8eb56d753d453d55692dd6532260876a6e461c

Reply:
@SSE4 looks like I've accidentally pushed it to my master, will fix this evening.

Reply:
@Talkless when fixing, prefer whitelist over blacklist, e.g.:
`if self.settings.os in ["Linux", "Android", "FreeBSD"]:`
is better than
`if not tools.is_apple_os(self.settings.os):`
because, there are other platforms which may use another extension than `.so` (e.g. Emscripten, at least)

Reply:
@SSE4 currently it's ` and self.settings.os in ("Android", "Linux"):`, will add FreeBSD too.

Reply:
@SSE4 pushed correctly now. Although that line now looks too long...